### ISSUE TITLE
Accessibility: SpeechNote doesn't work well with the Orca screen reader

### ISSUE THREAD

**Issue by devinprater** [mkiol/dsnote#168]
Hello,

I'm a blind person using Orca in order to use Linux. I'd like to use SpeechNote for work, in order to create audio files of material for others to use. 

Blind people use the keyboard to access computers, using a screen reader to speak what's going on as we type, tab around, all that. The screen reader for Linux is [Orca](https://help.gnome.org/users/orca/stable/index.html.en). Right now, SpeechNote doesn't work very well with Orca, having lots of unlabeled buttons, and I can't seem to activate any of the buttons on the screen.

I'm using Fedora Linux 40, Mate spin, running Orca 46.2 and I downloaded speechnote today, October 7, 2024. The issues I've found are:

* A lot of buttons are unlabeled. The button near "file," "languages," and the "notepad," and "translator" tabs is unlabeled. I'm not sure where it is onscreen so I'm telling you where about it is. The button near the text field is unlabeled as well, and I can't move passed that field either; I can only move backward around it. When I try to activate the File button (I assume this is the menu bar), nothing happens. Nothing happens also when I press Alt or F10 to activate the menu bar. Using Flat Review, a screen reader system to read things keyboard focus cannot review, I get the following:

```
Notepad Translator
Welcome!
push button File Languages
layered pane
separator
<p><i>Speech Note</i> let you take, read and translate notes in multiple languages. It uses Speech to Text, Text to Speech and Machine Translation to do so.</p><p>Text and voice processing take place entirely offline, locally on your computer, without using a network connection. Your privacy is always respected. No data is sent to the Internet.</p><p>To get started, you first need to set up the languages you want to use.</p><p>&rarr; Click the <i>Languages</i> button, select a language, and then download the model files you intend to use.</p><p>To switch between <i>Notepad</i> and <i>Translator</i> modes, use the toggle buttons in the upper right corner.</p><p>Have fun with <i>Speech Note</i>!</p>
separator
To speed up processing, enable hardware acceleration in the settings. push button
text
Neither Speech to Text nor Text to Speech model has been set up yet. Go to the <i>Languages</i> to download models for the languages you intend to use.
Close
Plain text push button
push button No Speech to Text model Listen separator push button No Text to Speech model x 1.0 Read
frame push button Cancel
```

Using the command to click on those items doesn't appear to do anything. The review contents don't change.

A few resources that may help:

* [tips for app developers](https://gitlab.gnome.org/GNOME/orca/-/blob/main/README-APPLICATION-DEVELOPERS.md?ref_type=heads)
* [QT accessibility docs](https://doc.qt.io/qt-6/accessible.html)
Please let me know if there's anything I can do to help. Thanks so much for this app, and I hope this helps bring it to more people!

**Comment by mkiol** [comment]
Hi, thank you for the report and for the hints.

I must admit that I was not aware of this problem. I'm just installing Orca and I promise to try to fix as much as possible for the next version.

**Comment by mkiol** [comment]
The new version 4.7.0 is out and available on flathub. 

The new version includes:

- Keyboard shortcuts enabled for several user interface elements. Elements such as menu items or buttons can be controlled using the keyboard shortcuts. Examples: Switch to Notepad (Ctrl+N), Switch to Translator (Ctrl+T), Open Languages (Ctrl+L), Read (Ctrl+Alt+Shift+R), Listen (Ctrl+Alt+Shift+L), Stop (Ctrl+Alt+Shift+S), Cancel (Ctrl+Alt+Shift+C), Pause (Ctrl+Alt+Shift+P)
- Buttons containing only icons have been extended with text labels

Unfortunately, it is still not possible to control Speech Note with just the keyboard. I tried to fix it but I didn't manage to do it completely. I found out that accessibility is a tricky thing and very hard. The main problem is that I have implemented custom UI components in many places in the UI and the built-in Qt accessibility module doesn't work well with it :(

Overall, the app should be more accessible, but it's still not possible to use it with just a keyboard and screen reader.

**Comment by devinprater** [comment]
Thank you so much for the progress! If I can listen to and export files, I think that's all I'd need for work purposes.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
